modules/detect_target/detect_target.py

```diff
@@ -75,6 +80,22 @@ def run(self, data: image_and_time.ImageAndTime) -> "tuple[bool, np.ndarray | No
                 assert detection is not None
                 detections.append(detection)
 
+        stop_time = time.time()
+
+        elapsed_time = stop_time - start_time
+      
+        for pred in predictions: 
+            with open('profiler.txt', 'a') as file:
+                speeds = pred.speed
```

Comment:
We don't want to add logic within the `detect_target.py` class for profiling. Rather we could time the worker outside of the call to detect target.

Reply:
The worker isn't the thing to test, something like this:
```py
# profiling_or_whatever_file_name.py

images: "list[np.ndarray]" = load_300_images()
detector = detect_target.DetectTarget(...)
for image in images:
    gc.disable()  # This disables the garbage collector
    start = time.time_ns()
    result, value = detector.run(image)  # Might or might not want to keep the bounding boxes
    end = time.time_ns()
    gc.enable()  # This enables the garbage collector
    if not result:
        # Handle error
  
    # Save results somewhere
    time_ns = end - start
    ...
```